internal/v2/tls_config_store/tls_config_store_test.go

```diff
@@ -216,30 +217,65 @@ func TestTLSConfigStoreServer(t *testing.T) {
 	stop()
 }
 
+func makeMap() map[commonpb.TLSVersion]uint16 {
+	m := make(map[commonpb.TLSVersion]uint16)
+	m[commonpb.TLSVersion_TLS_VERSION_1_0] = tls.VersionTLS10
+	m[commonpb.TLSVersion_TLS_VERSION_1_1] = tls.VersionTLS11
+	m[commonpb.TLSVersion_TLS_VERSION_1_2] = tls.VersionTLS12
+	m[commonpb.TLSVersion_TLS_VERSION_1_3] = tls.VersionTLS13
+	return m
+}
+
 func TestGetTLSMinMaxVersionsClient(t *testing.T) {
-	var tlsconfig s2av2pb.GetTlsConfigurationResp_ClientTlsConfiguration
-	tlsconfig.MinTlsVersion = commonpb.TLSVersion_TLS_VERSION_1_0
-	tlsconfig.MaxTlsVersion = commonpb.TLSVersion_TLS_VERSION_1_1
-	if _, _, err := getTLSMinMaxVersionsClient(&tlsconfig); err != nil {
-		t.Errorf("error in getTlsMinMaxVersionsClient logic: %v", err)
-	}
-	tlsconfig.MinTlsVersion = commonpb.TLSVersion_TLS_VERSION_1_1
-	tlsconfig.MaxTlsVersion = commonpb.TLSVersion_TLS_VERSION_1_0
-	if _, _, err := getTLSMinMaxVersionsClient(&tlsconfig); err == nil {
-		t.Errorf("getTlsMinMaxVersionsClient returned nil, should have returned error")
+	m := makeMap()
```

Comment:
Sorry for being unclear! What I was thinking was to have a test cases where we use:

1. `tlsConfig := &s2av2pb.GetTlsConfigurationResp_ServerTlsConfiguration {
				MinTlsVersion: commonpb.TLSVersion_TLS_VERSION_1_0 - 1,
				MaxTlsVersion: commonpb.TLSVersion_TLS_VERSION_1_3
			}`
2. `tlsConfig := &s2av2pb.GetTlsConfigurationResp_ServerTlsConfiguration {
				MinTlsVersion: commonpb.TLSVersion_TLS_VERSION_1_0,
				MaxTlsVersion: commonpb.TLSVersion_TLS_VERSION_1_3 + 1
			}`

Both of these examples should lead to errors. The rationale is: 1 is the defence-in-depth against the S2A returning a min version that is too low/invalid, and 2 is defence-in-depth against the S2A returning a max version that is too high/invalid.

Reply:
I see, thanks for elaborating! To catch this, I'll change the current logic of the function, to throw an error in the default case rather than just setting to tls.VersionTLS13. I'll also add a new TestXXX to test this scenario for each getTLSMinMaxVersionsClient/Server